internal/decoder/interface.go

```diff
@@ -308,7 +309,10 @@ func (d *interfaceDecoder) DecodeStream(s *Stream, depth int64, p unsafe.Pointer
 			*(*interface{})(p) = nil
 			return nil
 		}
-		return d.errUnmarshalType(rv.Type(), s.totalOffset())
+		t := rv.Type()
+		if t == errorType {
+			return d.errUnmarshalType(rv.Type(), s.totalOffset())
```

Comment:
Is it also doing encoding/json to return error only if an `error` interface type is implemented ?

Reply:
Yep.

Reply:
If so, please create a test case that implements the error interface for test coverage of this part